Plugin crashes with Gradle 8.7 and `-PcfLocal`

Consider an arbitrary plume-lib project.  For example:
```
git clone https://github.com/plume-lib/bibtex-clean.git
cd bibtex-clean
```

This command succeeds:
```
./gradlew wrapper --gradle-version 8.7 && ./gradlew build
```

But this command fails:
```
./gradlew wrapper --gradle-version 8.7 && ./gradlew build -PcfLocal
```

Here is the relevant part of the build failure:
```
FAILURE: Build failed with an exception.
* What went wrong:
Could not determine the dependencies of task ':shadowJar'.
> Failed to notify dependency resolution listener.
   > org/gradle/api/internal/artifacts/dependencies/DefaultSelfResolvingDependency
```

And here is the relevant part of the stack trace produced by `--stacktrace`:

```
Caused by: java.lang.NoClassDefFoundError: org/gradle/api/internal/artifacts/dependencies/DefaultSelfResolvingDependency
	at org.checkerframework.gradle.plugin.CheckerFrameworkPlugin$2$_beforeResolve_closure1$_closure2.doCall$original(CheckerFrameworkPlugin.groovy:217)
	at org.checkerframework.gradle.plugin.CheckerFrameworkPlugin$2$_beforeResolve_closure1$_closure2.doCall(CheckerFrameworkPlugin.groovy)
	at jdk.internal.reflect.GeneratedMethodAccessor187.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.api.specs.internal.ClosureSpec.isSatisfiedBy(ClosureSpec.java:33)
	at org.gradle.api.internal.collections.CollectionFilter.filter(CollectionFilter.java:49)
	at org.gradle.api.internal.collections.FilteredElementSource$FilteringIterator.findNext(FilteredElementSource.java:119)
	at org.gradle.api.internal.collections.FilteredElementSource$FilteringIterator.<init>(FilteredElementSource.java:112)
	at org.gradle.api.internal.collections.FilteredElementSource.iterator(FilteredElementSource.java:158)
	at org.gradle.api.internal.collections.FilteredElementSource.isEmpty(FilteredElementSource.java:82)
	at org.gradle.api.internal.DefaultDomainObjectCollection.isEmpty(DefaultDomainObjectCollection.java:335)
	at java_util_Set$isEmpty$0.call(Unknown Source)
	at org.checkerframework.gradle.plugin.CheckerFrameworkPlugin$2$_beforeResolve_closure1.doCall$original(CheckerFrameworkPlugin.groovy:214)
	at org.checkerframework.gradle.plugin.CheckerFrameworkPlugin$2$_beforeResolve_closure1.doCall(CheckerFrameworkPlugin.groovy)
	at jdk.internal.reflect.GeneratedMethodAccessor121.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.checkerframework.gradle.plugin.CheckerFrameworkPlugin$2.beforeResolve(CheckerFrameworkPlugin.groovy:210)
	at jdk.internal.reflect.GeneratedMethodAccessor188.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.event.DefaultListenerManager$ListenerDetails.dispatch(DefaultListenerManager.java:472)
	at org.gradle.internal.event.DefaultListenerManager$ListenerDetails.dispatch(DefaultListenerManager.java:454)
	at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:83)
	... 206 more
```

--- a/README.md
+++ b/README.md
@@ -12,7 +12,7 @@ Add the following to your `build.gradle` file:
 ```groovy
 plugins {
     // Checker Framework pluggable type-checking
-    id 'org.checkerframework' version '0.6.37'
+    id 'org.checkerframework' version '0.6.38'
 }
 
 apply plugin: 'org.checkerframework'
@@ -94,7 +94,7 @@ the definitions of the custom qualifiers.
 
 ### Specifying a Checker Framework version
 
-Version 0.6.37 of this plugin uses Checker Framework version 3.42.0 by default.
+Version 0.6.38 of this plugin uses Checker Framework version 3.42.0 by default.
 Anytime you upgrade to a newer version of this plugin,
 it might use a different version of the Checker Framework.
 
@@ -241,7 +241,7 @@ top-level project is a Java project).  For example:
 
 ```groovy
 plugins {
-  id 'org.checkerframework' version '0.6.37' apply false
+  id 'org.checkerframework' version '0.6.38' apply false
 }
 
 subprojects { subproject ->
@@ -282,7 +282,7 @@ plugins {
   id "net.ltgt.errorprone" version "1.1.1" apply false
   // To do Checker Framework pluggable type-checking (and disable Error Prone), run:
   // ./gradlew compileJava -PuseCheckerFramework=true
-  id 'org.checkerframework' version '0.6.37' apply false
+  id 'org.checkerframework' version '0.6.38' apply false
 }
 
 if (!project.hasProperty("useCheckerFramework")) {

--- a/build.gradle
+++ b/build.gradle
@@ -32,7 +32,7 @@ dependencies {
 }
 
 group 'org.checkerframework'
-version '0.6.37'
+version '0.6.38'
 
 gradlePlugin {
     website = 'https://github.com/kelloggm/checkerframework-gradle-plugin/blob/master/README.md'

--- a/src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy
+++ b/src/main/groovy/org/checkerframework/gradle/plugin/CheckerFrameworkPlugin.groovy
@@ -211,10 +211,13 @@ final class CheckerFrameworkPlugin implements Plugin<Project> {
           def depGroup = dependency.tokenize(':')[0]
           def depName = dependency.tokenize(':')[1]
           // Only add the dependency if it isn't already present, to avoid overwriting user configuration.
+          // The check for the current Gradle version was added because DefaultSelfResolvingDependency
+          // was removed in Gradle 8.7, but we still want to support not overwriting Checker Framework
+          // dependencies defined that way for earlier Gradle versions.
           if (project.configurations."$configuration.name".dependencies.matching({
             if (it instanceof DefaultExternalModuleDependency) {
               it.name.equals(depName) && it.group.equals(depGroup)
-            } else if (it instanceof DefaultSelfResolvingDependency) {
+            } else if (GradleVersion.current().compareTo(GradleVersion.version("8.7")) < 0  && it instanceof DefaultSelfResolvingDependency) {
               it.getFiles().any { file ->
                 file.toString().endsWith(depName + ".jar")
               }

